### PR TITLE
Try removing check_includes = false in //brave/browser/net

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/10659

As the target depends on `//brave/browser`, we don't want to have
explicit target for it, as it'll bring circular deps. Instead of that,
use `sources.gni` which directly added to the `//{chrome,brave}/browser` target.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc [hunk 1]
```diff
@@ -15,17 +15,17 @@
 #include "base/metrics/histogram_macros.h"
 #include "base/strings/string_util.h"
 #include "brave/browser/brave_browser_process.h"
-#include "brave/browser/brave_shields/ad_block_pref_service_factory.h"
-#include "brave/browser/brave_shields/brave_shields_web_contents_observer.h"
+#include "brave/browser/brave_shields/ad_block_pref_service_factory.h"  // nogncheck circular deps to //brave/browser
+#include "brave/browser/brave_shields/brave_shields_web_contents_observer.h"  // nogncheck circular deps to //brave/browser
```
Comment (sangwoo108) [inline browser/net/brave_ad_block_tp_network_delegate_helper.cc:19]: If we want to remove  // nogncheck and circular deps entirely, we would need to make a delegate interafce and remove direct access to the brave_shield and others.

Comment (sangwoo108) [reply browser/net/brave_ad_block_tp_network_delegate_helper.cc:19]: But I'm not sure if it's needed as //brave/browser and //chrome/browser seems equivalent layers. @goodov Do we need to split these into another target and remove circular deps?

File: browser/net/BUILD.gn [hunk 2]
```diff
@@ -22,9 +22,6 @@ buildflag_header("geolocation") {
 }
 
 source_set("net") {
```
Comment (goodov) [inline browser/net/BUILD.gn:24]: This explicit target makes no sense, because it uses things from `//chrome/browser` and `//brave/browser` a lot. Because of that, it should be a source list `//brave/browser/net/sources.gni` that must be included into `//brave/browser/sources.gni`.

`check_includes = false` and `// nogncheck` are basically the same thing, even if it only covers few files. `// nogncheck` may be used in some rare cases like adding an include that's not properly added on some platform or implicitly appears because of invalid GN dependency that's out of our control. Here we have a clear issue that should be solved with `source.gni` approach.

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc (hunk 1 above)
Comment (goodov) [reply browser/net/brave_ad_block_tp_network_delegate_helper.cc:19]: replied in a separate comment about those. We shouldn't add them and just should use `source.gni` here.

File: browser/net/BUILD.gn (hunk 2 above)
Comment (sangwoo108) [reply browser/net/BUILD.gn:24]: > This explicit target makes no sense, 

Yeah, perfectly makes sense. I'm going to remove `:net` target here and leave only test targets.

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc [hunk 3]
```diff
@@ -359,9 +360,9 @@ int OnBeforeURLRequest_AdBlockTPPreWork(const ResponseCallback& next_callback,
                                         std::shared_ptr<BraveRequestInfo> ctx) {
   // If the following info isn't available, then proper content settings can't
   // be looked up, so do nothing.
-  if (ctx->request_url.is_empty() ||
-      ctx->initiator_url.is_empty() || !ctx->initiator_url.has_host() ||
-      !ctx->allow_brave_shields || ctx->allow_ads ||
+  if (ctx->request_url.is_empty() || ctx->initiator_url.is_empty() ||
+      !ctx->initiator_url.has_host() || !ctx->allow_brave_shields ||
+      ctx->allow_ads ||
```
Comment (goodov) [inline browser/net/brave_ad_block_tp_network_delegate_helper.cc:365]: please squash commits and remove format-only changes from this PR.

File: browser/net/brave_block_safebrowsing_urls.cc [hunk 4]
```diff
@@ -16,15 +17,15 @@ namespace brave {
 const char kDummyUrl[] = "https://no-thanks.invalid";
 
 bool IsSafeBrowsingReportingURL(const GURL& gurl) {
-  static const std::vector<URLPattern> allowed_patterns({
+  static const base::NoDestructor<std::vector<URLPattern>> allowed_patterns({
```
Comment (goodov) [inline browser/net/brave_block_safebrowsing_urls.cc:20]: are these changes required for things to work or it's just to fix the codestyle? If it's the latter, I'd prefer to have this as a separate PR.

File: browser/sources.gni [hunk 5]
```diff
@@ -523,13 +525,13 @@ brave_chrome_browser_deps += brave_browser_permissions_deps
...
+brave_chrome_browser_deps += brave_browser_skus_deps
 brave_chrome_browser_deps += brave_browser_speedreader_deps
 brave_chrome_browser_deps += brave_browser_sync_deps
 brave_chrome_browser_deps += brave_browser_themes_deps
+brave_chrome_browser_deps += brave_browser_url_sanitizer_deps
 brave_chrome_browser_deps += brave_browser_web_discovery_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_prefs_deps
-brave_chrome_browser_deps += brave_browser_url_sanitizer_deps
-brave_chrome_browser_deps += brave_browser_brave_referrals_deps
```
Comment (goodov) [inline browser/sources.gni:536]: 👍

File: browser/net/brave_block_safebrowsing_urls.cc (hunk 4 above)
Comment (sangwoo108) [reply browser/net/brave_block_safebrowsing_urls.cc:20]: Unfortunately, this was needed as CI failed for 'declaration requires an exit-time destructor'

Comment (goodov) [reply browser/net/brave_block_safebrowsing_urls.cc:20]: I see. That's understandable then.

**Review by goodov — APPROVED** [review]

**Review by iefremov — APPROVED** [review]
thanks @sangwoo108 !
